template/pyproject.toml.jinja

```diff
@@ -1,16 +1,27 @@
 [build-system]
-{%- if backend == "hatch" %}
 requires = [
+{%- if backend == "hatch" %}
   "hatchling",
+{%- elif backend == "setuptools" %}
+  "setuptools",
+{%- endif %}
 {%- if backend == "hatch" and vcs == true %}
   "hatch-vcs",
+{%- elif backend == "setuptools" and vcs == true %}
+  "setuptools-scm",
 {%- endif %}
 ]
+{%- if backend == "hatch" %}
 build-backend = "hatchling.build"
+{%- elif backend == "setuptools" %}
+build-backend = "setuptools.build_meta"
 {%- endif %}
 
 [project]
 name = "{{ project_name }}"
+{%- if backend == "setuptools" %}
+version = "v0.1.0"
+{%- endif %}
```

Comment:
Is adding version here required?

Reply:
Yes the wheels weren't getting built without them, the version attribute should either be present in `pyproject.toml` or inside `setup.py`.

Reply:
I see, thanks for the clarification! In this case, should we add a condition to the `__version__` variable in `__init__.py` (to not be present when setuptools is selected)? I am a bit rusty on setuptools now lol. Does it automatically pick up this `version` (from `pyproject.toml`) and export it as `__version__` for the users?

Reply:
Yes I think we should remove the `__version__` variable from `__init.py__` as I think `scm` uses another `_version.pyi` file to keep track of the versions if I am not wrong?

Reply:
Not really; remember, `hatch-vcs` uses `setuptools-scm` internally, so most of the configuration options are the same. Users should always have the `__version__` variable. I just want to know if `setuptools` (without `scm`) does it on its own or should it be manual.

Reply:
> I see, thanks for the clarification! In this case, should we add a condition to the `__version__` variable in `__init__.py` (to not be present when setuptools is selected)?

> Users should always have the `__version__ `variable

I generally think it is okay to add it, but the use case for exporting a package's version through its API like this is diminishing nowadays, since one can now use `importlib.metadata.version("mypackage")` to programmatically access the version for a distribution package. However, one of the cases where adding it can be helpful is if the package's name differs from the top-level module or if it is under a namespace. For example, `importlib.metadata.version("PyYAML")` will return PyYAML's version, while PyYAML is imported in Python as `import yaml`.

So, I concur with @Saransh-cpp – we should indeed export `__version__` for the template/generated project to abstract this away from users, and we can use `importlib.metadata` to access it in `__init__.py`, unless @santacodes has some other thoughts.

```python
from importlib.metadata import version


__version__ = version("pybamm-example-project")
```

which will make it easier for someone to access the version of the package name if they were to modify the project slug manually.

> I am a bit rusty on setuptools now lol. Does it automatically pick up this `version` (from `pyproject.toml`) and export it as `__version__` for the users?

> I just want to know if `setuptools` (without `scm`) does it on its own or should it be manual.

No, it should be manual. While `setuptools.build_meta` can parse the version as a string from the abstract syntax tree of `mypackage.version` when installing the package to later export it as a part of its metadata (PEP 643), it would be strange for a build backend to modify a package's code and add an attribute to the package's public (or private) API.